killerbee/dev_rzusbstick.py

```diff
@@ -245,48 +259,81 @@ def get_dev_info(self):
                     usb.util.get_string(self.dev, self.dev.iProduct),     \
                     usb.util.get_string(self.dev, self.dev.iSerialNumber) ]
 
-    def __usb_write(self, endpoint, data):
+    def __usb_read(self):
+        '''
+        Read data from the USB device opened as self.handle.
+        
+        @rtype: String
+        @param data: The data received from the USB endpoint
+        '''
+        response = None
+        if USBVER == 0: # TODO: UNTESTED!!!!
+            try:
+                response = self.handle.bulkRead(RZ_USB_RESPONSE_EP, 1)[0]
+            except usb.USBError, e:
+                if e.args != ('No error',): # http://bugs.debian.org/476796
+                    raise e
+        else: #pyUSB 1.x
+            try:
+                response = self.dev.read(RZ_USB_RESPONSE_EP, self.dev.bMaxPacketSize0, timeout=500)
+                #print 'response length:', len(response)
+                #print response
+                #response = ''.join([chr(x) for x in response])
+                #response = response.pop()
+            except usb.core.USBError as e:
+                if e.errno != 110: #Not Operation timed out
+                    print "Error args:", e.args
+                    raise e
+                elif e.errno == 110:
+                    print "DEBUG: Received operation timed out error ...attempting to continue."
+        return response
+
+    def __usb_write(self, endpoint, data, expected_response=RZ_RESP_SUCCESS):
         '''
         Write data to the USB device opened as self.handle.
         
         @type endpoint: Integer
         @param endpoint: The USB endpoint to write to
+        @param expected_response: The desired response - defaults to RZ_RESP_SUCCESS
         @type data: Mixed
         @param data: The data to send to the USB endpoint
         '''
         if USBVER == 0:
             try:
                 self.handle.bulkWrite(endpoint, data)
                 # Returns a tuple, first value is an int as the RZ_RESP_* code
-                response = self.handle.bulkRead(RZ_USB_RESPONSE_EP, 1)[0]
+#                response = self.handle.bulkRead(RZ_USB_RESPONSE_EP, 1)[0]
             except usb.USBError, e:
                 if e.args != ('No error',): # http://bugs.debian.org/476796
                     raise e
-            time.sleep(0.0005)
-            if response != RZ_RESP_SUCCESS:
-                if response in RESPONSE_MAP:
-                    raise Exception("Error: %s" % RESPONSE_MAP[response])
-                else:
-                    raise Exception("Unknown USB write error: 0x%02x" % response)
+#            time.sleep(0.0005)
```

Comment:
Great change to keep read centralized. We should remove these commented out lines to make it clean. I can do that in a separate PR afterwards though if easiest.

Reply:
Yeah, I tend to leave them in particularly in other people's code so I'll let you decide what shold stay and what should go...